### PR TITLE
[Fluent 2 iOS] Merge Main

### PR DESCRIPTION
**Pull request by edjamesmsft**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Binary change:

| Before | After |
|--------|-------|
| 27,430,673 | 27,612,416 |

Merge main to fluent2-colors. This especially merges in the tokenized Button, which overwrote the fluent 2 colors change for Button. That will be fixed in another change in the future.

### Verification

Built and ran simulator.

|  Before (main)                         | Before (fluent2-colors)                        | After                                      |
|------------------------------|------------------------------|----------------------------|
| bottomSheetMainLight | bottomSheetColorLight | bottomSheetAfterLight |
| bottomSheetMainDark | bottomSheetColorDark | bottomSheetAfterDark |
| buttonMainLight | buttonColorLight | buttonAfterLight |
| buttonMainDark | buttonColorDark | buttonAfterDark |
| popupMainLight | popupColorLight | popupAfterLight |
| popupMainDark | popupColorDark | popupAfterDark |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1438)

### CODE REVIEW

File: ios/FluentUI/Popup Menu/PopupMenuItemCell.swift [hunk 1]
```diff
@@ -111,6 +111,13 @@ class PopupMenuItemCell: TableViewCell, PopupMenuItemTemplateCell {
         }
         updateColors()        // until popupmenuitemcell actually supports token system, clients will override colors via cell's backgroundColor property
         backgroundStyleType = .custom
+
+        tokenSetSink = tokenSet.sinkChanges { [weak self] in
+            guard let strongSelf = self else {
+                return
+            }
+            strongSelf.updateAppearance()
```
Comment (laminesm) [inline ios/FluentUI/Popup Menu/PopupMenuItemCell.swift:119]: We need to call updateColors after updateAppearance or it wouldn't fix the TVC overriding PopupMenu colors

File: ios/FluentUI.Demo/FluentUI.Demo/DemoListViewController.swift [hunk 2]
```diff
@@ -98,6 +98,8 @@ class DemoListViewController: DemoTableViewController {
         let demo = DemoControllerSection.allCases[indexPath.section].rows[indexPath.row]
         cell.setup(title: demo.title, accessoryType: .disclosureIndicator)
         cell.titleNumberOfLinesForLargerDynamicType = 2
+        cell.backgroundStyleType = .grouped
```
Comment (sophialee0416) [inline ios/FluentUI.Demo/FluentUI.Demo/DemoListViewController.swift:101]: Was your fluent2-branch updated to latest? I think I already merged this in last week: https://github.com/microsoft/fluentui-apple/pull/1416/files#diff-2792d90e0d200bfb9f58fd1846eb8141042fe4926b240a344893ef29038eae48R101

**Review by laminesm — APPROVED** [review]
Rip having to update button again

**Review by sophialee0416 — CHANGES_REQUESTED** [review]

**Comment by sophialee0416** [comment]
There's also some CCB changes that went into main that I don't happen to see here, could you bring those in as well?

File: ios/FluentUI.Demo/FluentUI.Demo/DemoListViewController.swift (hunk 2 above)
Comment (sophialee0416) [reply ios/FluentUI.Demo/FluentUI.Demo/DemoListViewController.swift:101]: Nvm please ignore I misread fluent2-colors, thought it was the tokens branch!

**Comment by edjamesmsft** [comment]
> (Editing the original comment): Are we bringing in the new CCB changes over here to fluent2-colors?

I will try to cherry-pick or merge CCB in again for this change, though it'll make it a little less clean of a merge!

File: ios/FluentUI/Popup Menu/PopupMenuItemCell.swift (hunk 1 above)
Comment (edjamesmsft) [reply ios/FluentUI/Popup Menu/PopupMenuItemCell.swift:117]: Fixed, confirmed with testing

File: ios/FluentUI/Command Bar/CommandBarButton.swift [hunk 3]
```diff
@@ -128,11 +132,10 @@ class CommandBarButton: UIButton {
         accessibilityLabel = (accessibilityDescription != nil) ? accessibilityDescription : title
         accessibilityHint = item.accessibilityHint
         accessibilityValue = item.accessibilityValue
-        accessibilityIdentifier = item.accessibilityIdentifier
     }
 
     private let isPersistSelection: Bool
-
+/*
```
Comment (sophialee0416) [inline ios/FluentUI/Command Bar/CommandBarButton.swift:138]: Why are these lines of code commented out?

Comment (sophialee0416) [reply ios/FluentUI/Command Bar/CommandBarButton.swift:138]: looks like not all the changes went in, sorry I might have missed out on why that is

Comment (edjamesmsft) [reply ios/FluentUI/Command Bar/CommandBarButton.swift:138]: I was moving to fast is why :) Will fix this momentarily!

Comment (sophialee0416) [reply ios/FluentUI/Command Bar/CommandBarButton.swift]: Totally understand! Just making sure :D

**Review by sophialee0416 — APPROVED** [review]

**Pull request merged**